### PR TITLE
[Core] Detect PointCollection changes in Polyline

### LINKED ISSUE #11848: Detect Points changes in Xamarin.Forms.Shapes.Polyline
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/1144380/detect-points-changes-in-xamarinformsshapespolylin.html)._

---
[severity:I'm frustrated, but able to complete my work]
Xamarin.Forms.Shapes.Polyline.Points doesn't respond to CollectionChanged events.
I bound the property Points of Polyline to my own PointCollection and when changing the points in the PointCollection the UI does not update the drawing with the new points. Even recreating a new PointCollection does not change the UI.
When I change the Xaml to use a Polygon instead of a Polyline everything works fine in Xamarin Forms 4.8.
This issue has been fixed in Xamarin Forms 4.8 for Polygon according to GitHub, and seems to work.
However, the issue still exist for Polyline. I even tried Xamarin Forms 5.0.1.1276-nightly and the issue is still there.
This is what I saw on GitHub
PolygonRenderer.cs Detect Points changes in Polygon (#11576) 12 days ago
PolylineRenderer.cs [Enhancement] Shapes (#9218) 2 months ago

Is there a plan to fix it for Polyline and when?

---
#### Denis Ruffo on 8/10/2020, 08:33 PM: 

<p>Polygon does update, however it would crash depending when the Point collection is updated.</p><p>because the PolygoneRenderer call for each platform (UWP, iOS, Android) the Method UpdatePoints without checking if Element is null. It is actually null for a while until not.</p><pre>void OnCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)    
{   UpdatePoints(); }

For example in ios
void UpdatePoints()
       {
               Control.UpdatePoints(Element.Points.ToCGPoints());
       }
Should be replaced with 
void UpdatePoints()
      {
          if (Element != null && Element.Points!=null)
          {
              Control.UpdatePoints(Element.Points.ToCGPoints());
          }
      }

There is a change  for PolyLine that has not been merged into Xamarin Forms 5.0
at https://github.com/xamarin/Xamarin.Forms/blob/polyline-points-changes/Xamarin.Forms.Platform.UAP/Shapes/PolylineRenderer.cs
and 
Xamarin.Forms/Xamarin.Forms.Platform.Android/Shapes/PolylineRenderer.cs
and 
Xamarin.Forms/Xamarin.Forms.Platform.iOS/Shapes/PolygonRenderer.cs

I created my own render based on this code and it seems to work, but also it suffers the same bug as PolylgoneRenderer in UpdatePoints. The code should check for Element not being null
void UpdatePoints()
      {
          if (Element != null && Element.Points!=null)
          {
              Control.UpdatePoints(Element.Points.ToCGPoints());
          }
      }

</pre>

#### Feedback Bot on 8/11/2020, 01:47 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Denis Ruffo on 8/11/2020, 02:29 PM: 

<p>In addition the previous issue.</p><p>if I have the following in my XAML</p><pre>&lt;Polyline 
 Points="{Binding DataPoints}"
 Stroke="Red" 
 StrokeDashArray="1,1"
 StrokeDashOffset="6"
 VerticalOptions="FillAndExpand"  
 StrokeThickness="2" /&gt;

</pre><p>The height of the display is scaled to the the biggest height when DataPoints was first created so if DataPoints has values with small amplitude and was originally initialized to small values like below </p><p>private PointCollection _dataPoints = new PointCollection();</p><p> for (int x = 0; x &lt; 100; x++)</p><p>     {       DataPoints.Add(new Point(x, 0));     } </p><p> and then later I have DataPoints with Amplitude of 100,  none of the data point would plot. </p><p>for (int x = 0; x &lt; 100; x++) </p><p>    {       DataPoints.Add(new Point(x, 100));     }</p><p> The only way I was able to fix it was to initialize DataPoints with the biggest amplitude possible  </p>

---
### Original Solutions
(no solutions)

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Detect PointCollection changes in Polyline (based on #11576).

### Fixes ###

- fixes #11848

### API Changes ###

 None

### Platforms Affected ### 

- Core/XAML (all platforms)

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Platform.Android/Shapes/PolylineRenderer.cs [hunk 1]
```diff
@@ -24,6 +25,9 @@ protected override void OnElementChanged(ElementChangedEventArgs<Polyline> args)
 
             if (args.NewElement != null)
             {
+                var points = args.NewElement.Points;
+                points.CollectionChanged += OnCollectionChanged;
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Platform.Android/Shapes/PolylineRenderer.cs:29]: aren't you supposed to unhook from the old element ?

Comment (StephaneDelcroix) [reply Xamarin.Forms.Platform.Android/Shapes/PolylineRenderer.cs:29]: and what if Points collection is fully replaced ?

Comment (Inrego) [reply Xamarin.Forms.Platform.Android/Shapes/PolylineRenderer.cs:29]: Both of your points are handled in the sample fix I provided in my original issue: https://github.com/xamarin/Xamarin.Forms/issues/11563#issuecomment-664039739

https://github.com/xamarin/Xamarin.Forms/pull/11576 seems to suffer from the same shortcomings.

**Comment by jsuarezruiz** [comment]
Rebased and fixed conflicts.

**Review by rmarinho — APPROVED** [review]

**Review by StephaneDelcroix — APPROVED** [review]

**Pull request merged**